ports/winit/app.rs

```diff
@@ -183,14 +337,45 @@ impl App {
         }
     }
 
+    /// Takes any outstanding events from the [Minibrowser], converting them to [EmbedderEvent] and
+    /// routing those to the App event queue.
+    fn queue_embedder_events_for_minibrowser_events(&self, minibrowser: &mut RefMut<Minibrowser>) {
+        for event in minibrowser.event_queue.borrow_mut().drain(..) {
+            match event {
+                MinibrowserEvent::Go => {
+                    let browser_id = self.browser.borrow().browser_id().unwrap();
+                    let location = minibrowser.location.borrow();
+                    let Ok(url) = ServoUrl::parse(&location) else {
+                        warn!("failed to parse location");
+                        break;
+                    };
+                    self.event_queue.borrow_mut().push(EmbedderEvent::LoadUrl(browser_id, url));
+                },
+            }
+        }
+    }
```

Comment:
Could this be a function on the Minibrowser struct that takes a mutable reference to the event queue? That would further isolate the minibrowser code in its own code unit.

Reply:
done in 06e0d2987e42a1fd542a34697be3f7ca37ecdc62